File: python/pages/forms_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class FormsPage(BasePage):

    NAME = (By.ID, "name")
    EMAIL = (By.ID, "email")
    SUBMIT_BTN = (By.ID, "submit-btn")
    RESULT = (By.ID, "result")

    def enter_name(self, name):
        self.type(self.NAME, name)
        return self

    def enter_email(self, email):
        self.type(self.EMAIL, email)
        return self

    def submit(self):
        self.click(self.SUBMIT_BTN)
        return self

    def get_result(self):
        return self.get_text(self.RESULT)

File: src/test/java/com/example/pages/FormsPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class FormsPage extends SeleniumBasePage {

    private final By nameField = By.id("name");
    private final By emailField = By.id("email");
    private final By submitBtn = By.id("submit-btn");
    private final By resultText = By.id("result");

    public FormsPage(WebDriver driver) {
        super(driver);
    }

    public FormsPage enterName(String name) {
        type(nameField, name);
        return this;
    }

    public FormsPage enterEmail(String email) {
        type(emailField, email);
        return this;
    }

    public FormsPage submit() {
        click(submitBtn);
        return this;
    }

    public String getResult() {
        return getText(resultText);
    }
}
